### PR TITLE
Fixed bug with iOS devices that crashed the post editor

### LINKED ISSUE #13223: `/new` composer page broken on iOS
**Describe the bug**

The composer page is broken in iOS... Noticed on Safari and within Forem app

**To Reproduce**

- Visit DEV on mobile
- Sign in
- Go to top-left menu and select "Write a Post"

**Expected behavior**

Page should load properly

**Screenshots**

Safari:

Image from iOS

Within Forem app webview:

IMG_0045

**Smartphone (please complete the following information):**

Observed by multiple iPhones.

Mine is v 14.4.2 and Forem app build 39.

**Additional context**

- This is not present on Forems that are not on the edge build (aka have not been deployed yet)
- Composing a post on DEV on mobile is most definitely not _primary behavior_ so we won't get as many reports on this as if it were desktop-wide.

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Fixed an issue where there was some kind of race condition with `.focus()` on some inputs within the mention autocomplete component that was causing browsers on iOS to crash.

## Related Tickets & Documents

Closes #13223 

## QA Instructions, Screenshots, Recordings

* Follow the QA instructions in #13101 (for the post editor) and #13061 (post comments)

### UI accessibility concerns?

N/A. The mention autocomplete retains the same level of accessibility it had prior to this change.

## Added tests?

- [ ] Yes
- [x] No, and this is why: The existing suite of tests covers these changes.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

Some US football player as a cartoon squishing a soccerball to look like a US football

### CODE REVIEW

File: app/javascript/crayons/MentionAutocompleteTextArea/MentionAutocompleteTextArea.jsx [hunk 1]
```diff
@@ -290,19 +298,17 @@ export const MentionAutocompleteTextArea = forwardRef(
...
-          replaceTextArea({
-            originalNodeToReplace: replaceElement,
-            plainTextArea,
-            comboboxTextArea,
-          });
-        }
-
-        // Initialize the new text areas in the "non-autosuggest" state, hiding the combobox until a search begins
-        comboboxTextArea.classList.add('hidden');
+      if (comboboxTextArea && plainTextArea && replaceElement) {
```
Comment (nickytonline) [inline app/javascript/crayons/MentionAutocompleteTextArea/MentionAutocompleteTextArea.jsx:301]: Focus is only required if we are replacing the element.

**Comment by nickytonline** [comment]
All the jest tests pass but there are three failure in Cypress tests all related to focus, which makes sense, as my changes are related to focus management. Oddly though in a browser outside of Cypress on my desktop and on iOS, the focus behaves as expected. Not sure why @aitchiss. Maybe it has to do with we're setting the focus programmatically and this throws off Cypress maybe, not really sure.

**Comment by nickytonline** [comment]
Ahh, so looks like focus stays on the textarea (plain input) when using the arrow keys and enter to select from the mention autocomplete (expected). So this looks to be a valid fail, but only in the scenario when a user clicks on an element in the mention autocomplete. Investigating how to handle that and still not have the crash on iOS.

**Comment by nickytonline** [comment]
@fdoxyz, if you have some time to test this tomorrow on an iOS device, that'd be awesome. I can provide you an URL on Slack to test it.

**Comment by fdoxyz** [comment]
Absolutely @nickytonline! I'll test this out as soon as I can and will let you know how it goes 💯

**Review by fdoxyz — APPROVED** [review]
Tested with iOS 14.4.2 in Safari + mobile app, Chrome + Safari on macOS 11.2.2 and Chrome on Android 10. All working great for me 🙌

**Review by aitchiss — APPROVED** [review]
I've tested this on desktop with Firefox/Safari/Chrome and all seems to be working well. I've tested on an Android device (on android 11.0) too and that all looks OK to me.

I think this is good to go 🚀

**Pull request merged**